### PR TITLE
IO-DEMO: register allocated memmory in memtrack

### PR DESCRIPTION
**Pull request by hoopoepg**
- register allocated memory in memtrack

### CODE REVIEW

File: test/apps/iodemo/ucx_wrapper.cc [hunk 1]
```diff
@@ -42,6 +44,38 @@ struct UcxAmDesc {
     const ucp_am_recv_param_t    *_param;
 };
 
+void* operator new(std::size_t size)
+{
+    void *ptr;
+
+    ptr = malloc(size);
+    ucs_memtrack_allocated(ptr, size, "io-demo operator new");
```
Comment (yosefe) [inline test/apps/iodemo/ucx_wrapper.cc:52]: "io_demo_new"

File: test/apps/iodemo/ucx_wrapper.cc [hunk 2]
```diff
@@ -42,6 +44,38 @@ struct UcxAmDesc {
...
+    ucs_memtrack_allocated(ptr, size, "io-demo wrapper malloc");
+
+    return ptr;
+}
+
+void UcxCallback::free(void *ptr)
+{
+    ucs_memtrack_releasing(ptr);
+    ::free(ptr);
+}
```
Comment (yosefe) [inline test/apps/iodemo/ucx_wrapper.cc:77]: 1. IMO malloc/free should be public static function in UcxContext
2. can we pass the allocation name as `const char *` to the malloc? MemoryPool already has a name, so we can use it

File: test/apps/iodemo/ucx_wrapper.cc (hunk 1 above)
Comment (hoopoepg) [reply test/apps/iodemo/ucx_wrapper.cc]: > "io_demo_new"

updated

File: test/apps/iodemo/ucx_wrapper.cc (hunk 2 above)
Comment (hoopoepg) [reply test/apps/iodemo/ucx_wrapper.cc]: > * IMO malloc/free should be public static function in UcxContext
> * can we pass the allocation name as `const char *` to the malloc? MemoryPool already has a name, so we can use it

done

File: test/apps/iodemo/io_demo.cc [hunk 3]
```diff
@@ -136,6 +136,12 @@ template<class BufferType, bool use_offcache = false> class ObjectPool {
         return UCS_MEMORY_TYPE_HOST;
     }
 
+
```
Comment (yosefe) [inline test/apps/iodemo/io_demo.cc:139]: extra space

Comment (hoopoepg) [reply test/apps/iodemo/io_demo.cc]: > extra space

removed

**Review by yosefe — APPROVED** [review]

**Comment by hoopoepg** [comment]
@yosefe ok to squash?

**Review by brminich — APPROVED** [review]

**Review by yosefe — APPROVED** [review]

**Comment by hoopoepg** [comment]
@yosefe ok to merge?

**Pull request merged**